### PR TITLE
feat(config): take config from env vars

### PR DESCRIPTION
**Pull request by ManuelBilbao**
**Motivation**

Currently we have many hardcoded literals in the code. With this PR the idea is to move that parameters to env vars.

**Description**

We'll use `envy` to load config from env vars directly to config structs that will be used to setup the components

### CODE REVIEW

File: crates/l2/operator/l1_tx_sender.rs [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+    contract_address: Address,
+    operator_address: Address,
+    operator_private_key: SecretKey,
+}
 
 impl L1TxSender {
-    async fn send_transaction(mut tx: EIP1559Transaction) -> Result<H256, String> {
-        let client = EthClient::new("http://localhost:8545");
-        let private_key = SecretKey::parse(&H256::from_str(RICH_WALLET_PK).unwrap().0).unwrap();
+    pub fn new_from_config(sender_config: L1TxSenderConfig, eth_config: EthConfig) -> Self {
```
Comment (ilitteri) [inline crates/l2/operator/l1_tx_sender.rs:22]: An alternative to this is to implement `From<L1TxSenderConfig> for L1TxSender`, and for all the components with the same behavior. It feels like the same for me, what do you prefer?

File: crates/l2/prover/sp1_prover.rs [hunk 2]
```diff
@@ -2,8 +2,7 @@ use tracing::info;
 
 use sp1_sdk::{ProverClient, SP1ProofWithPublicValues, SP1ProvingKey, SP1Stdin, SP1VerifyingKey};
 
-/// The ELF (executable and linkable format) file for the Succinct RISC-V zkVM.
-pub const FIBONACCI_ELF: &[u8] = include_bytes!("./sp1/program/elf/riscv32im-succinct-zkvm-elf");
+use crate::utils::config::prover::ProverConfig;
 
 pub struct SP1Prover {
```
Comment (ilitteri) [inline crates/l2/prover/sp1_prover.rs:7]: I know this is not in this PR's scope, but let's rename this to `Prover`

Comment (ilitteri) [reply crates/l2/prover/sp1_prover.rs:7]: Also the modules to `prover.rs`

File: crates/l2/utils/eth_client/engine.rs
Comment (ilitteri) [inline crates/l2/utils/eth_client/engine.rs:1]: Let's rename the module to `engine_client` and move it outside `eth_client` as they are different types of clients.

**Review by ilitteri — APPROVED** [review]
LGTM. Left a couple of comments, we can merge after addressing them.

File: crates/l2/prover/sp1_prover.rs (hunk 2 above)
Comment (ManuelBilbao) [reply crates/l2/prover/sp1_prover.rs]: Done

File: crates/l2/operator/l1_tx_sender.rs (hunk 1 above)
Comment (ilitteri) [reply crates/l2/operator/l1_tx_sender.rs:22]: Oh didn't notice this. Let's try to do a 1 to 1 mapping. In the case of the repeated config we can do something I'll suggest below.

Comment (ilitteri) [reply crates/l2/operator/l1_tx_sender.rs:22]: Nvm, we'll leave it for another instance

**Pull request merged**